frontend/tests/e2e/tests/AddVisaResidentWithPHNTest.js

```diff
@@ -0,0 +1,156 @@
+        .expect(AddVisaResidentWithPHNPage.ErrorText.nth(1).textContent).contains(IMMIGRATION_CODE_REQUIRED_MESSAGE)
+        .expect(AddVisaResidentWithPHNPage.ErrorText.nth(2).textContent).contains(PERMIT_ISSUE_DATE_REQUIRED_MESSAGE)
+        .expect(AddVisaResidentWithPHNPage.ErrorText.nth(4).textContent).contains(PERMIT_EXPIRY_DATE_REQUIRED_MESSAGE)
+        .expect(AddVisaResidentWithPHNPage.ErrorText.nth(5).textContent).contains(RESIDENCE_DATE_REQUIRED_MESSAGE)
+        .expect(AddVisaResidentWithPHNPage.ErrorText.nth(6).textContent).contains(COVERAGE_EFFECTIVE_DATE_REQUIRED_MESSAGE)
+        .expect(AddVisaResidentWithPHNPage.ErrorText.nth(7).textContent).contains(COVERAGE_CANCELLATION_DATE_REQUIRED_MESSAGE)
+        .expect(AddVisaResidentWithPHNPage.ErrorText.nth(8).textContent).contains(HOME_ADDRESS_REQUIRED_MESSAGE)
+        .expect(AddVisaResidentWithPHNPage.ErrorText.nth(9).textContent).contains(CITY_REQUIRED_MESSAGE)
+        .expect(AddVisaResidentWithPHNPage.ErrorText.nth(10).textContent).contains(PROVINCE_REQUIRED_MESSAGE)
+        .expect(AddVisaResidentWithPHNPage.ErrorText.nth(11).textContent).contains(POSTAL_CODE_REQUIRED_MESSAGE)
+});
+
+test('Check properly filled form passes validation', async t => {
+    await t
+        .typeText(PersonDetails.phnInput, '9882807277')
+        .click(PersonDetails.submitButton)
+        .wait(1000)
+        .setTestSpeed(.02)
+        // Given the page is filled out correctly
+        .typeText(AddVisaResidentWithPHNPage.groupNumberInput, '6337109')
+        .click(AddVisaResidentWithPHNPage.immigrationCodeSelect)
+        .click(immigrationCodeOption.withText("Student Authorization"))
+        .typeText(AddVisaResidentWithPHNPage.departmentNumberInput, '123456')
+        .typeText(AddVisaResidentWithPHNPage.visaIssueDateInput, '20210101')
+        .typeText(AddVisaResidentWithPHNPage.visaExpiryDateInput, '20221231')
+        .typeText(AddVisaResidentWithPHNPage.residenceDateInput, '20191108')
+        .typeText(AddVisaResidentWithPHNPage.coverageEffectiveDateInput, '20210301')
+        .typeText(AddVisaResidentWithPHNPage.coverageCancellationDateInput, '20211231')
+        .typeText(AddVisaResidentWithPHNPage.telephoneInput, '7802024022')
+        .typeText(AddVisaResidentWithPHNPage.address1Input, 'Test 111 ST')
+        .typeText(AddVisaResidentWithPHNPage.cityInput, 'VICTORIA')
+        .click(AddVisaResidentWithPHNPage.provinceSelect)
+        .click(provinceOption.withText('British Columbia'))
+        .typeText(AddVisaResidentWithPHNPage.postalCodeInput, 'V8V8V8')
+        .click(AddVisaResidentWithPHNPage.priorResidenceCodeInput)
+        .click(priorResidenceCodeOption.withText('British Columbia'))
+
+        // When I click the submit button
+		.click(AddVisaResidentWithPHNPage.submitButton)
+        // I expect a success message
+        .expect(AlertPage.alertBannerText.textContent).contains(SUCCESS_MESSAGE)
```

Comment:
Can we test the results as well? I.e. that the screen loads PHN and fullName>

Reply:
Might be bit tricky as same phn cannot be used again.

Reply:
Also, not sure if test HCIM is inline with test RAPID data